### PR TITLE
Optimize rendering

### PR DESCRIPTION
**Pull request by MrMeison**
- Add viewBox for responsibility for visual
- Block recalculate visual on resize
- Add label size settings
- remove unused code
- Calculate label size for font-size and inner radius
- Add editor config
- Update API to 1.6.0
- Update npm dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,26 +24,26 @@
...
     "powerbi-visuals-utils-tooltiputils": "0.3.0"
   },
   "devDependencies": {
     "@types/d3": "3.5.36",
     "@types/jasmine": "2.5.43",
     "@types/jasmine-jquery": "1.5.29",
-    "@types/jquery": "2.0.40",
-    "@types/lodash": "4.14.54",
-    "coveralls": "2.11.15",
+    "@types/jquery": "^2.0.41",
```
Comment (ignatvilesov) [inline package.json]: Please remove ```^```.

File: package.json [hunk 2]
```diff
@@ -24,26 +24,26 @@
...
+    "@types/lodash": "4.14.55",
+    "coveralls": "2.12.0",
     "jasmine": "2.5.3",
+    "jquery": "^3.2.0",
```
Comment (ignatvilesov) [inline package.json]: Please remove ```^```.

File: src/SunburstVisual.ts [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+                .classed(this.appCssConstants.main.class, true);
+
+            this.main = this.svg.append("g");
+            this.main.attr(CssConstants.transformProperty, translate(Sunburst.CentralPoint, Sunburst.CentralPoint));
+
+            this.selectedCategoryLabel = <d3.Selection<string>>this.svg
+                .append("text")
+                .classed(this.appCssConstants.label.class, true)
+                .classed(this.appCssConstants.categoryLabel.class, true);
+            this.selectedCategoryLabel.attr("text-anchor", "middle");
```
Comment (ignatvilesov) [inline src/SunburstVisual.ts]: Can we move it to the ```.less``` file? Is it possible?

File: src/SunburstVisual.ts [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+            this.svg.on("click", () => {
+                this.svg
+                    .classed(this.appCssConstants.mainInteractive.class, false);
+                this.labelsHidden = true;
+                this.selectionManager.clear();
+            });
+        }
+
+        public update(options: VisualUpdateOptions): void {
+            // if (options.type !== VisualUpdateType.Data && options.type !== VisualUpdateType.All) {
```
Comment (ignatvilesov) [inline src/SunburstVisual.ts]: Please remove this line. Nothing uses it.

File: src/SunburstVisual.ts [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+            if (options.type !== 2 && options.type !== 62) {
```
Comment (ignatvilesov) [inline src/SunburstVisual.ts]: Why do you use the magic numbers here? I suppose that we might use the enum instead.

File: src/SunburstVisual.ts [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+
+            return path;
+        }
+
+        private calculateLabelPosition(): void {
+            const innerRadius: number = Math.min.apply(
+                null,
+                this.data.root.children.map((x: SunburstSlice) => this.arc.innerRadius()(x, undefined))
+            );
+            const commonLineHeight: number = 1 + 1 * Sunburst.PercentageFontSizeMultiplier +
```
Comment (ignatvilesov) [inline src/SunburstVisual.ts]: Does ```1 * Sunburst.PercentageFontSizeMultiplier``` make sense? I suppose that we must remove the ```1``` here.

File: test/visualTest.ts [hunk 7]
```diff
@@ -24,96 +24,111 @@
...
 
-    // Sunburst1445472000808
-    import VisualClass = powerbi.extensibility.visual.Sunburst1445472000808.Sunburst;
-    import SunburstSlice = powerbi.extensibility.visual.Sunburst1445472000808.SunburstSlice;
-
-    const DefaultWaitForRender = 500;
-    const VisualSelector = ".sunBurstDrawArea";
+    const DefaultWaitForRender: number = 500;
+    const SliceSelector: string = ".sunburst__slice";
+    const LabelVisibleSelector: string = ".sunburst__label--visible";
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Is it BEM style selector?

Comment (MrMeison) [inline test/visualTest.ts]: Yes, it is BEM selectors

File: src/SunburstVisual.ts (hunk 6 above)
Comment (MrMeison) [inline src/SunburstVisual.ts]: `1` is `em` alternative here. the category label equals `1em` and the percentage label equals `2em`. Common width of labels - `1em` + `2em` + `line spacing (for this situation 1.2 em) `. Becouse `1` was added for more undenstanding

File: src/SunburstVisual.ts (hunk 5 above)
Comment (MrMeison) [inline src/SunburstVisual.ts]: I can't use enum for tests, because I use enum's number alternative.

File: src/SunburstVisual.ts (hunk 3 above)
Comment (MrMeison) [inline src/SunburstVisual.ts]: No, `text-anchor` is available only as SVG attribute

Comment (ignatvilesov) [inline src/SunburstVisual.ts]: I'm not so sure regarding you statement. Please take a look at [this](https://developer.mozilla.org/en/docs/Web/SVG/Attribute/text-anchor) document.

```
As a presentation attribute, it also can be used as a property directly inside a CSS stylesheet.
```

File: src/SunburstVisual.ts (hunk 6 above)
Comment (ignatvilesov) [inline src/SunburstVisual.ts]: ```1 * Sunburst.PercentageFontSizeMultiplier``` doesn't make sense. Please use ```Sunburst.PercentageFontSizeMultiplier``` instead. Please include a comment that describes the hidden logic.

File: test/visualTest.ts (hunk 7 above)
Comment (ignatvilesov) [inline test/visualTest.ts]: Sweet, but I'm not suse that it gives lots of profit right now.

File: package.json [hunk 8]
```diff
@@ -1,17 +1,17 @@
...
     "email": "pbicvsupport@microsoft.com"
   },
   "scripts": {
-    "postinstall": "pbiviz update 1.5.0",
+    "postinstall": "pbiviz update 1.6.0",
     "pbiviz": "pbiviz",
     "start": "pbiviz start",
     "package": "pbiviz package",
-    "lint": "node node_modules/tslint/bin/tslint -r \"node_modules/tslint-microsoft-contrib\" \"+(src|test)/**/*.ts\"",
+    "lint": "tslint \"+(src|test)/**/*.ts\"",
```
Comment (AviSander) [inline package.json]: why revert this change

File: src/SunburstVisual.ts [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+            this.svg.on("click", () => {
+                this.svg
+                    .classed(this.appCssConstants.mainInteractive.class, false);
+                this.labelsHidden = true;
+                this.selectionManager.clear();
+            });
+        }
+
+        public update(options: VisualUpdateOptions): void {
+            if (options.type !== 2 && options.type !== 62) {
```
Comment (AviSander) [inline src/SunburstVisual.ts]: can you 'const' this so we know the meanings of '2' and '62'
